core/stopWatch.go

```diff
@@ -9,32 +9,33 @@ import (
 // MeasurementsLoggerFormat contains the formatting string to output elapsed time in seconds in a consistent way
 const MeasurementsLoggerFormat = "%.4fs"
 
-type stopWatch struct {
-	mut         sync.Mutex
+// StopWatch is used to measure duration
+type StopWatch struct {
```

Comment:
It has no exported fields, indeed, but it has exported methods.

If all code that uses the **stop watch in a function signature (as parameter or as return value)**  would re-define the interface - if we also stick to `define interface where it is needed / consumed` - a bit too much / overkill. 

Encapsulation not really violated. Interface not necessary. Rarely mocked. Those who mock it can define interface. Also, `StopWatch` is not defined as abstraction in major frameworks.

Reply:
ok